android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/notifications/ScopedNotificationRequest.java

```diff
@@ -0,0 +1,54 @@
+package versioned.host.exp.exponent.modules.universal.notifications;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import expo.modules.notifications.notifications.interfaces.NotificationTrigger;
+import expo.modules.notifications.notifications.model.NotificationContent;
+import expo.modules.notifications.notifications.model.NotificationRequest;
+import host.exp.exponent.kernel.ExperienceId;
+
+public class ScopedNotificationRequest extends NotificationRequest {
+  // We store String instead of ExperienceId, cause ScopedNotificationRequest must be serializable.
+  private String mExperienceIdString;
+
+  public ScopedNotificationRequest(String identifier, NotificationContent content, NotificationTrigger trigger, @Nullable ExperienceId experienceId) {
+    super(identifier, content, trigger);
+    if (experienceId == null) {
+      mExperienceIdString = "";
+    } else {
+      mExperienceIdString = experienceId.get();
+    }
+  }
+
+  private ScopedNotificationRequest(Parcel in) {
+    super(in);
+    mExperienceIdString = in.readString();
+  }
+
+  boolean checkIfBelongsToExperience(@Nonnull ExperienceId experienceId) {
+    if (mExperienceIdString == null) {
+      return false;
```

Comment:
Hm… if no `experienceId` has been provided… maybe we should deliver the notification to all the experiences? 🤔 I can imagine two scenarios in which `experienceId` in a notification could be `null`:
- a developer has their own custom development client ([this one](https://docs.expo.io/guides/adhoc-builds/)), they have access to device push token, they send notification to the device push token, they shouldn't need to worry about `experienceId` (or should they? maybe it depends on what they use the client for? but overall if they test app's notifications by device push token on a development client maybe they shouldn't need to change the notification-sending code between sending the notifications to standalone apps (no `experienceId` required [OH, it shouldn't be required in standalone apps, **this** it the reason we want to deliver them to all experiences]) and the client?)
- the Expo team sends a notification to all developers, like eg. _A new SDK is released!_ in which case… we could send it with `experienceId: @expo/home`?